src/main/java/appeng/block/misc/BlockSkyCompass.java

```diff
@@ -164,4 +186,31 @@ public void addCollidingBlockToList( final World w, final BlockPos pos, final Ax
 	{
 
 	}
+
+	@Override
+	public EnumBlockRenderType getRenderType( IBlockState state )
+	{
+		return EnumBlockRenderType.ENTITYBLOCK_ANIMATED;
+	}
+
+	@Override
+	@SideOnly( Side.CLIENT )
+	public TileEntitySpecialRenderer<? extends AEBaseTile> getTESR()
+	{
+		return new SkyCompassTESR();
+	}
+
+	@Override
+	@SideOnly( Side.CLIENT )
+	public ItemMeshDefinition getItemMeshDefinition()
+	{
+		return is -> new ModelResourceLocation( "appliedenergistics2:tile.BlockSkyCompass", "normal" );
```

Comment:
There is already a good amount of mods requiring Java8. So it is more or less already a requirement for modded minecraft.

It certainly makes many things easier. Even java7 with `try-with-resources` could make some things easier. As these are one of the few things actually causing memory leaks and even forge is unable to use them `try` correctly with native resources.

The only counter-argument I currently know comes from modpack makers, Who are complaining about their players being too stupid to install or upgrade java and they would face more support requests.
But then we make the mod for fun, we do not need to sell it to users and keep the userbase as large as possible. So even when it means losing 5% or 10% of them, it is not really an issue for us, if it allows us to spend time more efficiently. Even as business it should really be considered dropping some percent of the userbase, if it means you can save time/costs or improve it for the majority of your users compared to still have to maintain legacy stuff.